internal/user/repository/eventsourcing/model/email.go

```diff
@@ -0,0 +1,92 @@
+package model
+
+import (
+	"encoding/json"
+	"github.com/caos/logging"
+	"github.com/caos/zitadel/internal/crypto"
+	caos_errs "github.com/caos/zitadel/internal/errors"
+	es_models "github.com/caos/zitadel/internal/eventstore/models"
+	"github.com/caos/zitadel/internal/user/model"
+	"time"
+)
+
+type Email struct {
+	es_models.ObjectRoot
+
+	EmailAddress    string `json:"email,omitempty"`
+	IsEmailVerified bool   `json:"-"`
+
+	isEmailUnique bool `json:"-"`
+}
+
+type EmailCode struct {
+	es_models.ObjectRoot
+
+	Code   *crypto.CryptoValue `json:"code,omitempty"`
+	Expiry time.Duration       `json:"expiry,omitempty"`
+}
+
+func (e *Email) Changes(changed *Email) map[string]interface{} {
+	changes := make(map[string]interface{}, 1)
+	if changed.EmailAddress != "" && e.EmailAddress != changed.EmailAddress {
+		changes["email"] = changed.EmailAddress
+	}
+	return changes
+}
+
+func EmailFromModel(email *model.Email) *Email {
+	return &Email{
+		ObjectRoot:      email.ObjectRoot,
+		EmailAddress:    email.EmailAddress,
+		IsEmailVerified: email.IsEmailVerified,
+	}
+}
+
+func EmailToModel(email *Email) *model.Email {
+	return &model.Email{
+		ObjectRoot:      email.ObjectRoot,
+		EmailAddress:    email.EmailAddress,
+		IsEmailVerified: email.IsEmailVerified,
+	}
+}
+
+func EmailCodeToModel(code *EmailCode) *model.EmailCode {
+	return &model.EmailCode{
+		ObjectRoot: code.ObjectRoot,
+		Expiry:     code.Expiry,
+		Code:       code.Code,
+	}
+}
+
+func (u *User) appendUserEmailChangedEvent(event *es_models.Event) error {
+	u.Email = new(Email)
+	u.IsEmailVerified = false
```

Comment:
false is set by default
```suggestion
	u.Email = new(Email)
```